packages/admin/admin/src/dataGrid/usePersistentColumnState.tsx

```diff
@@ -80,7 +80,8 @@ export function usePersistentColumnState(stateKey: string): GridProps {
         [mediaQueryColumnVisibilityModel, setStoredColumnVisibilityModel],
     );
 
-    const [pinnedColumns, setPinnedColumns] = useStoredState<GridPinnedColumns>(`${stateKey}PinnedColumns`, {});
+    const [pinnedColumns, setPinnedColumns] = useStoredState<GridPinnedColumns>(`${btoa(window.location.pathname)}${stateKey}PinnedColumns`, {});
```

Comment:
The same issue probably occurs with the column visibility above. 
Maybe create a prefix for all `useStoredState()` instances 🤔 

```ts
const storageKeyPrefix = `${btoa(window.location.pathname)}${stateKey}`;
// ...
const [storedColumnVisibilityModel, setStoredColumnVisibilityModel] = useStoredState<GridColumnVisibilityModel>(`${storageKeyPrefix}ColumnVisibility`, {});
const [pinnedColumns, setPinnedColumns] = useStoredState<GridPinnedColumns>(`${storageKeyPrefix}PinnedColumns`, {});
```

Reply:
Yes, it should be for all stored states in this file